woocommerce/woocommerce-ios
# Order Creation: Add shipping row

Closes: https://github.com/woocommerce/woocommerce-ios/issues/6025.

## Description

This PR adds "Shipping" actionable row to the "Payment" section in the new order screen.

## Changes

- Adds new row in `OrderPaymentSection` behind feature flag.
- Adds `ShippingLine?` property to `OrderDetails` of view model.
- Adds 2 properties to `PaymentDataViewModel`.
- Updates `PaymentDataViewModel` sync flow to use `ShippingLine` in "order total" calculation and update new shipping row properties.
- Adds `SelectionStyle` to `TitleAndValueRow` to apply different button styles.

## Testing

1. Build and run the app in debug mode (to ensure the feature flag is enabled).
2. Make sure Order Creation is enabled under Settings > Experimental Features.
3. Go to the Orders tab and tap the + button. Tap "Create order" in the bottom sheet.
4. Tap "Add Product" button and select any product to enable "Payment" section.
5. Tap "Add shipping" button to add a hardcoded $10 shipping line.
6. Confirm that the "Order Total" is correctly updated to display products + shipping price.
7. Tap "Shipping" row to remove the shipping line.
8. Confirm that the "Order Total" is correctly updated to display products price.

## Screenshots

Add shipping | Shipping line
--|--
![Simulator Screen Shot](https://user-images.githubusercontent.com/3132438/153918746-c5cb85ef-4b46-4417-bf5f-ce6b644c88be.png)|![Simulator Screen Shot](https://user-images.githubusercontent.com/3132438/153918783-72f51bd6-e491-4f87-82b2-25b62da65f90.png)

---
